cas-server-webapp/src/main/webapp/WEB-INF/web.xml

```diff
@@ -110,7 +110,8 @@
         </servlet-class>
         <init-param>
             <param-name>contextConfigLocation</param-name>
-            <param-value>/WEB-INF/cas-servlet.xml, /WEB-INF/cas-servlet-*.xml</param-value>
+            <!-- Load the child application context. Start with the default, then modules, then overlays. -->
+            <param-value>/WEB-INF/cas-servlet.xml,classpath*:/META-INF/cas-servlet-*.xml,/WEB-INF/cas-servlet-*.xml</param-value>
```

Comment:
Exactly. If you need access to the root spring application context, you need to add your XML config file to the META-INF/spring directory. If you need access to the servlet child web application context, simply create `cas-servlet-oauth-oidc.xml` and CAS will pick up. Note that the servlet is a child of the parent, and while you can reach up to the parent from the child, you cant do the opposite. So you may need java classes to handle bean registrations in a dynamic way.